operational/compute_instance_migration/risc-compute-instance-migration-recos.pt

```diff
@@ -152,56 +161,151 @@ script "js_formatStacks", type: "javascript" do
+      lowcost.push({
+        provider: target.provider,
+        location: target.location,
+        cost: target.cost,
         provider_name: target.provider_name,
         provider_id: target.provider_id,
         stack_name: stack["stack_name"],
         stackid: stack["stackid"]
       })
-    }
-  })
+    })
+    //Find the overall lowest price
+    var lowest = _.min(lowcost, function(provider_cost){ return provider_cost.cost; });
 
+    if (lowest.cost != null){
+      var lowest_cost = lowest.cost; 
+    } 
+
+    var result_rec = {}
+    result_rec["Stack Name"] = stack["stack_name"];
+    _.each( lowcost, function(cost) {
+      var highlight = "";
+      if (cost.cost == lowest_cost) highlight = "***";
+      if (cost.cost !==  100000) {
+        result_rec[cost.provider + " location"] = (cost.location.length > 1) ? highlight + cost.location + highlight : '';
+        result_rec[cost.provider + " cost"] = highlight + "$" + parseFloat(cost.cost).toFixed(2) + highlight;        
+      } else {
+        result_rec[cost.provider + " location"] = '';
+        result_rec[cost.provider + " cost"] = '$0';  
+      }
+    })
+
+    result.data.push(result_rec);
+    result.header = _.keys(result_rec);
+  })
 
 EOS
 end 
 
+script "parse_report", type: "javascript" do
+  parameters "data"
+  result "result"
+  code <<-EOS
+    var result = {};
+    result.sResult = "";
+
+    console.log(data.header);
+
+    //Function repeats a character a number of times
+    function pad(input, times) {
+        sRet = "";
+        for (i=0; i < times; i++) sRet += input;
+        return sRet;
+    }
+
+    //format the Report output
+    result.sResult = "| ";
+    var sSeparator = "| ";
+    for (x=0; x < data.header.length; x++) {
+      result.sResult += data.header[x] + " | ";
+      sSeparator += pad('-',data.header[x].length) + " | ";
+    }
+    result.sResult = result.sResult.trim() + "\\n";
+    result.sResult += sSeparator.trim() + "\\n";
+    
+    //now the data
+    for (i=0; i < data.data.length; i++) {
+      if (data.data[i][data.header[0]] != "") {
+        sLine = "| ";
+        for (x=0; x < data.header.length; x++) {
+          sLine += data.data[i][data.header[x]] + " | ";
+        }
+        result.sResult += sLine.trim() + "\\n";     
+      } 
+    }
+  EOS
+end
+
+datasource "output_report" do 
+  run_script $parse_report, $ds_formatStacks
+end
+
 policy "stack_migration_recos" do
-  validate_each $ds_formatStacks do
+  validate $output_report do
     summary_template "Application Migration Recommendations"
     detail_template <<-EOS
 
-| Application Stack | Recommended Provider/Region | Estimated Hourly Cost | Stack ID | Provider ID |
-| ----------------- | --------------------------- | --------------------- | -------- | ----------- |
-{{ range data -}}
-| {{.stack_name}} | {{.provider_name}} | {{.cost}} | {{.stackid}} | {{.provider_id}} |
-{{ end -}}
+{{data.sResult}}
```

Comment:
This is nice, but I'd prefer to keep the other method as it's easier to see the table layout.

Reply:
I understand, but as the table has a dynamic number of columns, this was the only reason.. We could change it to the new Incident data structure if you like and it is available .
 MArkdown had  a dynamic times of

Reply:
Dynamic isn't going to work well, especially with the new export block for the resource table.  Is there a way to normalize the data or just show blank values in columns

Reply:
you can use the new export block for the resource table but we won't merge until the UI is released to support the new table, by EOM.

Reply:
I believe we should leave it as is now, and then rebuild the table once we have the new UI

Reply:
I disagree.  The resource table isn't going to be dynamic.  Why worry about changing it later, and since you are the one who knows the code, it will be hard to get the same structure later.

Reply:
I don't want to have to rely on you to update it later or have nimbus try to figure out you dynamic table.